clientapi/routing/routing.go

```diff
@@ -117,15 +117,16 @@ func Setup(
 		).Methods(http.MethodGet, http.MethodPost, http.MethodOptions)
 	}
 
-	r0mux := publicAPIMux.PathPrefix("/r0").Subrouter()
+	v3mux := publicAPIMux.PathPrefix("/{version:(?:r0|v3)}/").Subrouter()
```

Comment:
Can we please just stick a comment here explaining the regex? 

> "make a named path var called version and give it the matcher for a non-capturing group (because everything in the path var is 'captured' so having additional named capture groups makes no sense) which then selects r0 OR v3, with a trailing slash to stop v333 matching